pkgs/by-name/ri/rip2/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "rip2";
+  version = "0.9.0";
+  src = fetchFromGitHub {
+    owner = "MilesCranmer";
+    repo = pname;
+    rev = version;
+    hash = "sha256-9leLWfPilDQHzQRzTUjAFt9olTPEL4GcQgYFWZu3dug=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
Why is this instead of `cargoHash`?

Reply:
Saw some other packages do this. Wasn’t sure what the best practice was

Reply:
Currently we insist on using `cargoHash` wherever possible (upstream has `Cargo.lock`, and no Git dependencies) because of the overhead of evaluation and repo size brought by cargo lock. This may change in the future as neither is a completely ideal solution

Reply:
Got it, makes sense